src/utils/fuse.ts

```diff
@@ -1,12 +1,13 @@
 import { FuseResult } from 'fuse.js'
 
-import { SearchMatch } from '@/types/fuse'
+import { Match } from '@/types/fuse'
 
 type ResultWithMatch<T> = T & {
-  matches: SearchMatch[]
+  matches: Match[]
 }
 
 export function withMatches<T>(result: FuseResult<T>): ResultWithMatch<T> {
+  // @ts-expect-error assigning readonly indices to mutable
```

Comment:
Using `ReadonlyArray` instead of `Array` in /src/types/fuse.ts` solves this problem